x/wasm/keeper/relay.go

```diff
@@ -169,6 +170,15 @@ func (k Keeper) OnRecvPacket(
 		// submessage errors result in error ACK with state reverted. Error message is redacted
 		return nil, err
 	}
+
+	if data == nil {
+		// contract wants async acknowledgement, so store the packet for later
+		err = k.StoreAsyncAckPacket(ctx, convertPacket(msg.Packet))
```

Comment:
I moved this call into the keeper now (was in IBCHandler before). That makes the whole thing a little easier to understand, but now we have to convert the packet back into a `channeltypes.Packet`